test/lifecycle_module_test.dart

```diff
@@ -32,6 +32,7 @@ class MockStreamSubscription extends Mock implements StreamSubscription<Null> {}
 class TestLifecycleModule extends LifecycleModule {
   Iterable<StreamSubscription<LifecycleModule>> _eventListStreamSubscriptions;
   bool _managedDisposerWasCalled = false;
+  String _name;
```

Comment:
Oh, see, I was hoping for a named constructor on the `LifecycleModule` class. I don't think consumers should have to override a private attribute to set their own name.